audformat/core/database.py

```diff
@@ -463,9 +466,14 @@ def map_files(
         """
         def job(table):
             if table.is_segmented:
-                table.df.index = table.df.index.map(
+                index = table.df.index.map(
                     lambda x: (func(x[0]), x[1], x[2])
                 )
+                index = utils.set_index_dtypes(
```

Comment:
After rebasing I got an error because `utils.union()` is no more restrict about the dtypes and calling `map()` apparently changed the dtype of the file level from 'string' to 'object'